recipes/autoconf/all/conanfile.py

```diff
@@ -9,30 +11,31 @@ class AutoconfConan(ConanFile):
     description = "Autoconf is an extensible package of M4 macros that produce shell scripts to automatically configure software source code packages"
     topics = ("conan", "autoconf", "configure", "build")
     license = ("GPL-2.0-or-later", "GPL-3.0-or-later")
-    exports_sources = "patches/**"
     settings = "os", "arch", "compiler"
+
+    exports_sources = "patches/*"
     _autotools = None
 
     @property
     def _source_subfolder(self):
         return os.path.join(self.source_folder, "source_subfolder")
 
-    def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        os.rename("{}-{}".format(self.name, self.version), self._source_subfolder)
+    def build_requirements(self):
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/cci.latest")
+        self.build_requires("m4/1.4.18")
```

Comment:
```suggestion
        if hasattr(self, "settings_build"):
            self.build_requires("m4/1.4.18")
```

Reply:
1 profile doesn't like to see the same dependency in requirements and build requirements. When versions are the same, all is fine, but it can quickly break (if the version of overridden by downstream).

Reply:
This is working fine on 1, or 2 profile builds.
It's the environment variables that are not propagated ('AUTOCONF`) to the test package.

I will apply your suggestion nonetheless.

Reply:
It works fine in a unit test, but may not work when integrated in a complex dependency graph. Try to create a recipe that would require another version of m4 than the one required by autoconf, and autoconf => it breaks with 1 profile.

Reply:
Oh, version clashes of build and host requirements. That's right, I have encountered those. Hard to resolve with conan.

Reply:
Can you please take a look at https://github.com/conan-io/conan-center-index/issues/6341#issuecomment-880862257?
This pr is part of fixing #6341, but I feel conan is sabotaging us.